distribution/lib/Standard/Base/0.0.0-dev/src/System/File/Generic/File_Like.enso

```diff
@@ -1,4 +1,5 @@
 import project.Data.Text.Text
+import project.Enso_Cloud.Data_Link.Data_Link_From_File
```

Comment:
It is weird that `System.File.Generic` depends on `Enso_Cloud`. It should be the opposite way. This seems [braided like spaghetti](http://wiki.apidesign.org/wiki/Use_Modular_Architecture).

Reply:
Summarizing our discussions, the user facing `File` concept is designed to follow data links in some operations (e.g. `size`, `is_directory`, `read`). Thus it needs to depend on some data link module.

~~I think introducing an SPI for it would be over-engineering it because there can be at most one implementation of this SPI.~~ Well after some thought I retract this. 

Technically we _could_ introduce a `LinkSPI` that would allow specifying custom behaviour on certain kinds of file-links, and then having only one implementation of that (`Data_Link`).

Alternatively, we could introduce `System_File` concept that is the low-level direct mapping of `TruffleFile` and then keep `File` as the high-level user-facing concept that is wrapping a `System_File`.

Regardless of that - we can indeed move the `Data_Link` module out from the `Cloud` module. It is there because from product perspective, Data Links are part of our Cloud offering. From system-design perspective, `Cloud` _uses_ the `Data_Link` module but the `Data_Link` module does not need to depend on `Cloud` so it _can_ be separated. That does not change the dependency between `File` and `Data_Link` though.